Sources/Machine/rbm_spin.cc

```diff
@@ -167,25 +195,35 @@ RbmSpin::VectorType RbmSpin::LogValDiff(
   const std::size_t nconn = tochange.size();
   VectorType logvaldiffs = VectorType::Zero(nconn);
 
-  thetas_ = (W_.transpose() * v + b_);
+  thetas_ = detail::DenseFwd(W_, v, b_);
   RbmSpin::lncosh(thetas_, lnthetas_);
 
   Complex logtsum = lnthetas_.sum();
 
-  for (std::size_t k = 0; k < nconn; k++) {
-    if (tochange[k].size() != 0) {
-      thetasnew_ = thetas_;
-
-      for (std::size_t s = 0; s < tochange[k].size(); s++) {
-        const int sf = tochange[k][s];
-
-        logvaldiffs(k) += a_(sf) * (newconf[k][s] - v(sf));
-
-        thetasnew_ += W_.row(sf) * (newconf[k][s] - v(sf));
+  if (a_.has_value()) {
```

Comment:
There is basically no measurable speed-up gain in separating the two branches instead of just putting the bias to zero